### PR TITLE
Ignore TABLE_SCAN_MAXMEM for root table

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

**Comment by dlmarion** [comment]
I'm curious if we should ignore this property for all tables in the accumulo namespace for proper function of the system.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java [hunk 1]
```diff
@@ -76,6 +77,11 @@ public void run() {
       TableConfiguration acuTableConf = server.getTableConfiguration(session.threadPoolExtent);
       long maxResultsSize = acuTableConf.getAsBytes(Property.TABLE_SCAN_MAXMEM);
 
+      // Always return large root tablets
+      if (session.threadPoolExtent.tableId().equals(AccumuloTable.ROOT.tableId())) {
+        maxResultsSize = Long.MAX_VALUE;
```
Comment (dlmarion) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java:82]: The property default is 512K, so maybe Long.MAX_VALUE is not the right choice here

Comment (cshannon) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java:82]: I agree that going smaller than Long.MAX_VALUE makes sense.

Maybe the best thing is to just take the maximum value between the default and the user setting so we always use at least the default size but allow users to go higher.

```java
maxResultsSize = Math.max(maxResultsSize, ConfigurationTypeHelper
  getFixedMemoryAsBytes(Property.TABLE_SCAN_MAXMEM.getDefaultValue()));
```

**Review by cshannon — COMMENTED** [review]
I think overall the strategy makes sense to have some sensible minimum value (probably the default) for the system tables when scanning.

We could also improve the logging around the memory check to make the issue more obvious but just not letting it easily happen in the first place makes sense to me.

Something else I'm wondering is if we should throw an error if the config is bad. So if the default config (site config, etc) is set lower than it can just be ignored for the system tables. But if a user specifically tries to set a lower value on a system table like metadata then maybe we should throw an exception and prevent that config change as it would just be ignored anyways.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java]: Implemented in 13249fb

**Comment by dlmarion** [comment]
>We could also improve the logging around the memory check to make the issue more obvious but just not letting it easily happen in the first place makes sense to me.
> But if a user specifically tries to set a lower value on a system table like metadata then maybe we should throw an exception and prevent that config change as it would just be ignored anyways.

I'm wondering if instead of changing the code to throw an exception, we just document the behavior change in Property.java. Something like `Values lower than the default value are ignored for system tables.`

**Comment by cshannon** [comment]
> I'm wondering if instead of changing the code to throw an exception, we just document the behavior change in Property.java. Something like `Values lower than the default value are ignored for system tables.`

That's probably fine too if it is documented so that users are at least aware it will be ignored and it isn't unexpected.

**Comment by dlmarion** [comment]
Closing as #4457 fixes the root cause.

**Pull request closed without merging**